lib/group.c

```diff
@@ -931,15 +932,21 @@ grn_table_group_single_key_records(grn_ctx *ctx, grn_obj *table,
     grn_obj *real_key = key;
     data.need_resolved_id = false;
     if (key->header.type == GRN_ACCESSOR) {
-      grn_accessor *a = (grn_accessor *)key;
-      if (a->action == GRN_ACCESSOR_GET_KEY &&
-          a->next && a->next->action == GRN_ACCESSOR_GET_COLUMN_VALUE &&
-          a->next->obj && !a->next->next) {
-        real_key = a->next->obj;
-        data.need_resolved_id = true;
+      grn_accessor *a;
+      for (a = (grn_accessor *)key;
+           a->next;
+           a = a->next) {
+        if (a->action != GRN_ACCESSOR_GET_KEY) {
+          break;
+        }
+        if (a->next->action == GRN_ACCESSOR_GET_COLUMN_VALUE &&
+            a->next->obj && !a->next->next) {
+          real_key = a->next->obj;
+          data.need_resolved_id = true;
+        }
```

Comment:
Can we simplify this?

e.g.:

```c
      grn_accessor *a;
      for (a = (grn_accessor *)key;
           a->action == GRN_ACCESSOR_GET_KEY && a->next;
           a = a->next) {
        /* do nothing */
      }
      if (a->next &&
          a->next->action == GRN_ACCESSOR_GET_COLUMN_VALUE &&
          a->next->obj &&
          !a->next->next) {
        real_key = a->next->obj;
        data.need_resolved_id = true;
      }
```

Reply:
Yes. `grn_obj_column()` returns a column itself not an accessor for the column with a column name of a table.